src/jit/lsra.cpp

```diff
@@ -5723,7 +5723,7 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
-        if (isAssigned(availablePhysRegInterval ARM_ARG(currentInterval->registerType)))
+        if ((intervalToUnassign != nullptr) && isAssigned(availablePhysRegInterval ARM_ARG(currentInterval->registerType)))
         {
             intervalToUnassign = availablePhysRegInterval->assignedInterval;
```

Comment:
As I see we did not use the value of `intervalToUnassign ` before. We always rewrite it with `availablePhysRegInterval->assignedInterval` or did not use at all.

Can we replace it with like `bool needUnassign = false` use it in the condition and declare 'intervalToUnassign' in this if.

Reply:
> As I see we did not use the value of `intervalToUnassign` before. We always rewrite it with `availablePhysRegInterval->assignedInterval` or did not use at all.

`intervalToUnassign` was used before the changes in https://github.com/dotnet/coreclr/commit/62ef107532203badddded0cc7f7a06765afffa7a which introduced this regression.

> Can we replace it with like `bool needUnassign = false` use it in the condition and declare `intervalToUnassign` in this if.

Sure, that's fine.